app/lib/frontend/backend.dart

```diff
@@ -658,11 +659,15 @@ Future<models.PackageVersion> parseAndValidateUpload(
 /// Automatically detects the built-in types.
 /// The returned list should be alphabetically sorted.
 List<String> detectTypes(Pubspec pubspec) {
-  final List<String> detectedTypes = [];
+  final Set<String> detectedTypes = new SplayTreeSet();
```

Comment:
Usually we don't use the splay data structures much -- they also might go away soon. I'd just use the normal `Set()`.